internal/provider/generators/shared/codegen/emitter.go

```diff
@@ -392,57 +392,73 @@ func (e Emitter) emitAttribute(tfType string, attributeNameMap map[string]string
 				validatorsGenerator = stringValidators
 
 			case cfschema.PropertyTypeObject:
-				if len(property.Items.PatternProperties) > 0 {
-					return features, unsupportedTypeError(path, "list of key-value map")
-				}
-
 				if len(property.Items.Properties) == 0 {
 					return features, unsupportedTypeError(path, "list of undefined schema")
-				}
-
-				e.printf("schema.ListNestedAttribute{/*START ATTRIBUTE*/\n")
-				e.printf("NestedObject: schema.NestedAttributeObject{/*START NESTED OBJECT*/\n")
-				e.printf("Attributes:")
-
-				f, err := e.emitSchema(
-					tfType,
-					attributeNameMap,
-					parent{
-						computedAndOptional: computedAndOptional,
-						computedOnly:        computedOnly,
-						path:                path,
-						reqd:                property.Items,
-					},
-					property.Items.Properties)
-
-				if err != nil {
-					return features, err
-				}
+				} else if len(property.Items.PatternProperties) > 0 {
+					// List of map
+					var nestedPatternProperty *cfschema.Property
+					for _, v := range property.Items.PatternProperties {
+						nestedPatternProperty = v
+						break
+					}
+					if nestedPatternProperty == nil {
+						return features, unsupportedTypeError(path, "list of key-value map with no pattern")
+					}
+					switch nestedPatternProperty.Type.String() {
```

Comment:
nit - some visual spacing to separate the nil check from the switch on pattern type.

```suggestion
					}

					switch nestedPatternProperty.Type.String() {
```